cgt_calc/parsers/trading212.py

```diff
@@ -98,32 +105,57 @@ def __init__(self, header: list[str], row_raw: list[str], filename: str):
         self.exchange_rate = decimal_or_none(row["Exchange rate"])
-        self.transaction_fee = decimal_or_none(row.get("Transaction fee (GBP)", "0"))
-        self.finra_fee = decimal_or_none(row.get("Finra fee (GBP)", "0"))
-        self.stamp_duty = decimal_or_none(row.get("Stamp duty (GBP)", "0"))
-        self.conversion_fee = decimal_or_none(
-            row.get("Currency conversion fee (GBP)", "0")
-        )
-        fees = (
-            (self.transaction_fee or Decimal(0))
-            + (self.finra_fee or Decimal(0))
-            + (self.conversion_fee or Decimal(0))
-        )
+        self.transaction_fee = Decimal(row.get("Transaction fee (GBP)") or "0")
+        transaction_fee_foreign = Decimal(row.get("Transaction fee") or "0")
+        if transaction_fee_foreign > 0:
+            if row.get("Currency (Transaction fee)") != "GBP":
+                raise ParsingError(
+                    filename,
+                    "The transaction fee is not in GBP which is not supported yet",
+                )
+            self.transaction_fee += transaction_fee_foreign
+        self.finra_fee = Decimal(row.get("Finra fee (GBP)") or "0")
+        self.stamp_duty = Decimal(row.get("Stamp duty (GBP)") or "0")
+        self.conversion_fee = Decimal(row.get("Currency conversion fee (GBP)") or "0")
+        conversion_fee_foreign = Decimal(row.get("Currency conversion fee") or "0")
+        if conversion_fee_foreign > 0:
+            if row.get("Currency (Currency conversion fee)") != "GBP":
+                raise ParsingError(
+                    filename,
+                    "The transaction fee is not in GBP which is not supported yet",
+                )
+            self.conversion_fee += conversion_fee_foreign
+        fees = self.transaction_fee + self.finra_fee + self.conversion_fee
         if "Total" in row:
             amount = decimal_or_none(row["Total"])
             currency = row["Currency (Total)"]
         else:
             amount = decimal_or_none(row["Total (GBP)"])
             currency = "GBP"
+        if (
+            amount is not None
+            and (action == ActionType.BUY or self.raw_action == "Withdrawal")
+            and amount > 0
+        ):
+            amount *= -1
         price = (
-            abs(amount / quantity)
+            abs((amount + fees) / quantity)
             if amount is not None and quantity is not None
             else None
         )
-        if amount is not None:
-            if action == ActionType.BUY or self.raw_action == "Withdrawal":
-                amount *= -1
-            amount -= fees
+        reported_price = self.price_foreign
+        if (
+            reported_price is not None
+            and price is not None
+            and (self.currency_foreign == "GBP" or self.exchange_rate is not None)
+        ):
+            reported_price /= self.exchange_rate or Decimal("1")
+            assert abs(reported_price - price) < Decimal("0.01"), (
```

Comment:
That could be a reason why you had mismatch here.

Reply:
The reason there's a mismatch is because trading212 rounds the price / share.
Even when checking the transaction in the UI I get this discrepancy:
Sell 24 shares at $164.01 with exchange rate 1.25229 and £4.71 fees. Trading212 total says £3138.50. However,
24*164.01/1.25229-4.71=£3138.52

I believe there might be some implicit rounding both in the csv and in the UI. Some of them I was able to fetch more decimals from the UI and updating the csv. Others are unfixable, but since is just a few cents is not that important. I set it to a warning now.
Ultimately, what matters is the £££ received/sent, which is what I'm using here.